### ISSUE TITLE
Disqualified shares opening phase

### ISSUE THREAD

**Issue by nkuba** [keep-network/keep-core#378]
Implement DKG protocol [Phase10]() and shares opening part of [Phase 11](http://docs.keep.network/cryptography/beacon_dkg.html#_phase_11_disqualified_share_reconstruction).

Depends on #318 , #312

---

### Phase 10: Disqualified share opening

All active players in _G_10_ broadcast the keys they share with
players in _DQ_9_, so the reconstruction of Pedersen-VSS can be done
offline.

``` python
# Calculate:

G_10 = G_9 - DQ_9

keys_i = []

for m in DQ_9:
  keys_i += X_im

# Broadcast:

if keys_i /= []:
  keys_i
```

### Phase 11: Disqualified share reconstruction

Decrypt and reconstruct _z_m_ for every participant _P_m_ that presented
valid shares in _phase 4_ but whose public key shares in _phase 7_ were invalid.

```python
# Receive:

[keys_1, keys_2, ... , keys_N]

# Calculate:

(G_11, IA_10, DQ_10) = checkBroadcast(10, G_10)

for m in (G_6 - G_11): # <1>
  for j in G:
    X_jm = keys_j[m]

    K_jm = ecdh(X_jm, Y_mj)

    (s_mj, t_mj) = decrypt(K_jm, E_mj)

  ss_m = take(T + 1, [s_m1, ... , s_mN])

  is_m = [s.index for s in ss_m]

# CONTINUATION IN ANOTHER TASK
```
Continued in #320.
